extensions/amp-animation/0.1/runners/native-web-animation-runner.js

```diff
@@ -125,11 +140,18 @@ export class NativeWebAnimationRunner extends AnimationRunner {
     });
   }
 
+  /**
+   * Pause if runner was initialized.
+   */
+  maybePause() {
+    this.isInitialized_() && this.pause();
+  }
```

Comment:
1. Isn't this exactly the same as `pause()` with the introduced short-circuit?
2. In which cases are we calling this when uninitialized? Are we masking over a race?

Reply:
> story pausing with long-press will call `maybePause` here, but there are no animations here to pause

In that case, we should not call `pause` nor `resume` from the Story side with `prefers-reduced-motion`. Would that allow you to remove the `isInitialized_()` checks here?

Reply:
I would say that it would somewhat help but there's an edge case that I didn't want to mention, which is that users can change the `reduced-motion` while the story is already loaded, and that could break the story. This can happen when you change the reduced motion settings and come back to the story page after a couple of hours / days if the page was loaded , so it's sth that could happen (not many times but still I'd want to not break the story if it happens). Also we don't know if an OS ever wants to turn on `Reduced Motion` with low battery or other things like that.

Eg:
1. Load story with `prefers-reduced-motion` (animations don't get initialized)
2. Turn off `Reduced Motion`
3. Pause the story with long-press (which would call `pause` and `resume`)

A fix would be to cache the `reduced-motion` value when a page loads, or add a check (which is what I'm doing now). I prefer adding a check, but don't feel strongly, so LMK if you think otherwise.

Reply:
> This can happen when you change the reduced motion settings and come back to the story page after a couple of hours / days if the page was loaded

This should be rare, but in any case the `amp-animation` changes should _not_ mask over this.

Either:

1. Persist the `reduced-motion` value as you mentioned, or
2. (Preferrable) Make sure that Story pages update their motion settings correctly when returning to the document. This is easily done in a `resumeCallback()`.

Reply:
Makes sense. Currently we're handling the logic of the `prefers-reduced-motion` on the `AnimationRunner` so it would be a good idea to keep it there. Adding a method on the `resumeCallback` of `AmpStoryPage` would need to call `AnimationRunner.reinitializeOrFinalizeAnimation()` which feels like it's introducing extra complexity, besides the `resumeCallback` could have a race with the `AmpStoryPage.resume()` which is called in a wide range of ocassions and also is bound to `onVisibilityChanged`.

So, for the sake of simplicity and maintainability (and stability), `#1` sounds like a better alternative.

Reply:
Ok, then back to the goal of persisting the value. We should now undo the changes to `native-web-animation-runner.js`, since we now expect calls in a correct order:

1. Remove local `isInitialized_()` checks.
2. Remove `maybe*` methods.
3. Same as above, on `.externs.js`